### DISCUSSION TITLE
Building levels hesitation

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#3295] [Feedback]
I don't think there's anything to do about this, but I just wanted to share that despite all the [**work**](https://github.com/streetcomplete/StreetComplete/issues/3092) we've put into clarifying the building levels quest, stuff like this still trips me up a bit:

![ee81fbb6-4d09-4c46-983e-f445faf3a017](https://user-images.githubusercontent.com/6563664/133868868-a82c370f-c32f-4bab-921c-10f3f2957c27.jpg)

I took this picture intending to open an issue about "how do I tag this?", but upon reflection I'm pretty sure it should be 2 levels. You can see if you walk around the left side that the garage part is fully above-ground, not a basement level. But at the same time, it's not *really* part of the house… Usually I decide "basement" based on where the foundation is, but this is like an above-ground/foundation basement, or something.

**Comment by andrewharvey** [comment]
Same here, I used to tag these as 2 but now tag as 1, and still don't know what's correct, so all my building levels answers of either 1 and 2 aren't really reliable. This needs to be solved on the wiki first, only then can SC improve the quest to ensure it's accurately tagged.

**Comment by Helium314** [comment]
If you can enter the building at the bottom level, I would definitely answer 2 levels (in the photo I only see windows and no door).

There are some other situations unclear to me where you have something like [this](https://westnordost.de/p/34246.jpg), but with a door leading from outside to the half-cellar level (steps inside the building). I'm never sure whether to tag such cases as 1 or 2 levels, and my answer usually depends on how far below ground level the inside is.

**Reply by smichel17** [reply]
In the [other thread](https://github.com/streetcomplete/StreetComplete/issues/3092) (where I posted that image), I *think* we decided that if it's a basement level, it should not be counted, even if there's some windows peaking out from underground. So that picture (which, let me just upload it here so it is not deleted when the note is closed) would be tagged as one level.

![image](https://user-images.githubusercontent.com/6563664/133886790-9d6f5ef2-9555-47c9-aaeb-1b6684e6d21b.png)

**Reply by smichel17** [reply]
> in the photo I only see windows and no door

That's funny, when I walked by I could have sworn that there was a garage door. But now that I look at the picture it seems that was not the case. In my imagination, it was like this:

![phantom-garage-door](https://user-images.githubusercontent.com/6563664/133886927-f2d48df0-eee2-47f2-a498-354563921174.png)

Note that if you walk around the corner on the left, it is finished with the same type of siding and windows as the floor above; it is almost certainly fully above the foundation on the left side. I'm not sure why the windows are so high up on the front. Maybe the foundation comes up half the height of the wall on that side? :shrug:

**Reply by matkoniecz** [reply]
> That's funny, when I walked by I could have sworn that there was a garage door.

I was 100% sure about when looking at image - likely due to driveway reaching it and garage style windows.

**Comment by smichel17** [comment]
Tangential thought on building levels that I'll just leave here: I was noticing yesterday that it's a quest which is easier to solve at an angle from the house. If you are right in front of a house, as in the picture here, and the roof is sloped, you often can't tell whether there is a level in it or not until you walk a little further and can see if there is a window.

**Comment by mnalis** [comment]
I'd say, **do not waste much time thinking it over - just use one or the other reasoning, or even intermix them** - it does not matter, as others will estimate otherwise anyway .

As mentioned before in https://github.com/streetcomplete/StreetComplete/issues/3092#issuecomment-886758232, there will **always** be (at least) that `+/-1` error in `building:levels`, even if we all somehow magically find and agree on ideal solution, simple because of the tag inertia and people using old definitions, so that particular tag is goner as far as precision is required, not matter the amount of brain power invested to it here (or whereever).
And even if we invented a new `building:precise_levels` tag with ultra-precise rules, take note that not all floors needs to have windows even if there are building levels inside, and outsider would still need to estimate, which is again prone to error (or, much more likely, refrain from almost ever answering such quest which requires 20 pages of instructions with illustrations and examples) 

Also, likely the only thing this key is somewhat useful for is making 3d representation of the city, for which it is also quite unadequate for any amount of precision ([height](https://wiki.openstreetmap.org/wiki/Key:height) in meters is a little more useful, and then mostly only when you've split `building` into several `building:part` pieces; see https://wiki.openstreetmap.org/wiki/Simple_3D_Buildings. And this is what even this **simple** 3D model wants you to map every building in much more detail: 
![roof_diagram](https://user-images.githubusercontent.com/156656/133905755-60aed4b0-e5e8-4fa3-935a-09d4ed8587b4.jpg)
And that is quite simple, compared to [more advanced 3D model](https://wiki.openstreetmap.org/wiki/F3DB) (and abandoned, for good reason)

So only mapping `building:levels` on `building=house` seems to fall quite short for even that simple 3D model. When you add that unavoidable `+/- 1` levels (or so) error, I firmly believe that anyone using flight simulator over terrain generated from OSM data won't be horribly offended if that `building:levels=3` building is actually 4-storey building, or vice versa. They'd (maybe) be mildly interested (for few seconds) in how much that 20-level tower in the middle of the city is contrasted to all that 3-4 storey buildings around it, but that is about it.
